src/test/json/Object_test.cpp

```diff
@@ -209,13 +209,17 @@ class JsonObject_test : public ripple::test::TestOutputSuite
 
     void testKeyFailure ()
     {
-#ifdef DEBUG
         setup ("repeating keys");
         auto& root = makeRoot();
         root.set ("foo", "bar");
         root.set ("baz", 0);
-        auto fail = [&]() { root.set ("foo", "bar"); };
-        expectException (fail);
+        // setting key again throws in !NDEBUG builds
+        auto set_again = [&]() { root.set ("foo", "bar"); };
+#ifdef NDEBUG
```

Comment:
Why switch from `DEBUG` to `NDEBUG`?

Reply:
the code that is being tested is predicated on `NDEBUG`:  https://github.com/ripple/rippled/blob/710d9b6538f7537cf5f61fdec207e94748e14329/src/ripple/json/impl/Writer.cpp#L150  ..so I figured we should at least align with that. Also, whereas `NDEBUG` is fairly standard (assert depends on it), `DEBUG` is not and we might want consider removing it from our macros.

Reply:
I missed the symmetry when I was digging into the writer. That's an excellent point.